main/eval/src/mill/eval/GroupEvaluator.scala

```diff
@@ -342,6 +315,20 @@ private[mill] trait GroupEvaluator {
           )
         }
       }
+
+      // Remove any empty `.dest/` folders to tidy up the filesystem. Otherwise
+      // tasks that call `os.proc` or `T.dest` without actually doing anything with
+      // the folder will leave empty folders around cluttering the disk
+      this.synchronized {
+        for {
+          p <- usedDest
+          if os.list.stream(p).headOption.isEmpty
+          // workers can be used after they return their value and
+          // may need to continue working with their `T.dest`,
+          if !isWorker
+        } os.remove(p)
+      }
+
```

Comment:
I think this could break tasks, that export their `dest` dir in on purpose. E.g. I use this sometimes to refer to working directories used by other task. Also the `GenIdea` uses empty `dest` dirs as unique `output` dirs when configuring the Java/Scala compiler.

https://github.com/com-lihaoyi/mill/blob/6e35f6e56145da74f11252c1ded78f08b8a9964c/scalalib/src/mill/scalalib/GenIdeaModule.scala#L34-L36

Reply:
Yeah i noticed that. The use cases i hit was in 11-module-run-task, and in how we used to save a folder for running Mill's own integrstion tests. Both were hacks and i found solutions. We always tell people not to write to upstream dest folders after all.

Could there be another way of marking those tasks that need the folder? e.g. putting a marker file in the folder like git requires (https://www.theserverside.com/), or defining another variant of `Task` that is just a scratchspace that preserves the folder for usage

Reply:
another alternstive is to create a folder inside .dest and export that instead, just to tell Mill that you have something in dest tou want to keep

Reply:
decided to revert this for now, can discuss it in a follow up